src/xml/reader.cr

```diff
@@ -50,11 +46,12 @@ class XML::Reader
       nil,
       options
     )
+    LibXML.xmlTextReaderSetStructuredErrorHandler(@reader, ->Error.structured_callback, Box.box(@errors))
   end
 
   # Moves the reader to the next node.
   def read : Bool
-    collect_errors { LibXML.xmlTextReaderRead(@reader) == 1 }
+    LibXML.xmlTextReaderRead(@reader) == 1
```

Comment:
I should double check the xmlTextReader API to verify there isn't rogue `xmlGenericError` calls (libxml < 2.13).